library/src/main/java/com/novoda/downloadmanager/lib/DownloadExecutorFactory.java

```diff
@@ -0,0 +1,48 @@
+package com.novoda.downloadmanager.lib;
+
+import android.content.Context;
+import android.content.pm.ApplicationInfo;
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+
+import com.novoda.notils.logger.simple.Log;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+class DownloadExecutorFactory {
+    private static final int KEEP_ALIVE_TIME = 10;
+    private static final int DEFAULT_MAX_CONCURRENT_DOWNLOADS = 5;
+    private static final String METADATA_MAX_CONCURRENT_DOWNLOADS = "com.novoda.downloadmanager.MaxConcurrentDownloads";
+
+    public ExecutorService createExecutor(Context context) {
+        int maxConcurrentDownloads = getMaximumConcurrentDownloads(context);
+        ThreadPoolExecutor executor = new ThreadPoolExecutor(
```

Comment:
Looking at the latest Lollipop update, it seems that they haven't used the `newFixedThreadPool` (http://androidxref.com/5.1.0_r1/xref/packages/providers/DownloadProvider/src/com/android/providers/downloads/DownloadService.java#120), but we can take a look to see if there's any difference between the two implementations.